README.md

```diff
@@ -20,29 +20,24 @@ material library would go in the `assets/material/` directory and be at
 All asset files should be under the `assets` directory in an appropriate
 subdirectory.
 
-## Optimization
+## Creating images
 
-Please consider optimization tools for assets.
+Images must be code-generated, see the `packages/diagrams/lib/src/` directory.
 
-For PNGs, we recommend `optipng`, using the following command line:
+## Generation
 
-```bash
-optipng -zc1-9 -zm1-9 -zs0-3 -f0-5 *.png
-```
+See the [`generate.dart`](./bin/generate.dart) script. It will
+regenerate almost all of existing assets using the Flutter version you
+have installed.
 
-Be careful about applying this aggressively. In particular, files in
-the `assets/tests` directory should not be optimized.
-
-The automatic generation tool will automatically apply optimization to
-the assets it generates.
+Typically, you plug in an Android phone and run:
 
-## Generation
+```sh
+PATH=~/dev/flutter/bin/cache/dart-sdk/bin:~/Android/Sdk/platform-tools:$PATH bin/generate.sh
+```
 
-See the [documentation for the generate.dart script in the `diagrams`
-directory](packages/diagrams/README.md), in conjunction with the
-[`generate.dart`](./bin/generate.dart) script. It will regenerate almost all of
-existing assets using the Flutter version you have installed. Feel free
-to add more modules in the `diagrams` package to generate new assets.
+You may wish to add the `--category` argument to focus just on the
```

Comment:
You might mention `--name`, which is almost more useful than `--category`.

Reply:
I couldn't figure out how to use `--name`.

Reply:
`--category` can narrow down the generation script to a library of diagrams, while `--name` can drill down on a specific diagram. Every `DiagramMetadata` implementation has this name property which also serves as the file name. 
[Example Here](https://github.com/flutter/assets-for-api-docs/blob/951c8c526cc57c9634088971ae56c5197f65dc99/packages/diagrams/lib/src/switch_list_tile.dart#L188)